File: solutions_5658571765186560_0/java/nonpop/Codejam4.java
package codejam4;

import java.util.Scanner;

public class Codejam4 {

    public static void main(String[] args) {
        
        Scanner s = new Scanner(System.in);
        int testCases = s.nextInt();
        for (int testCase = 1; testCase <= testCases; testCase++) {
            int X = s.nextInt();
            int R = s.nextInt();
            int C = s.nextInt();

            boolean richardWins;
            if (R*C % X != 0) {
                richardWins = true;
            } else {
                switch(X) {
                    case 1:
                    case 2:
                        richardWins = false;
                        break;
                    case 3: 
                        richardWins = (R == 1 || C == 1);
                        break;
                    case 4: 
                    case 5: 
                        richardWins = (R <= 2 || C <= 2);
                        break;
                    case 6: 
                        richardWins = (R <= 3 || C <= 3);
                        break;
                    default:
                        richardWins = true;
                        break;
                }
            }

            if (richardWins)
                System.out.println("Case #" + testCase + ": RICHARD");
            else
                System.out.println("Case #" + testCase + ": GABRIEL");
        }
    }
    
}

File: solutions_5658571765186560_1/C/Lkillua/codejam4.c
#include<stdio.h>

int main()
{
  int t,q=1,x,r,c,p;
  scanf("%d",&t);
  while(t--)
  {
    scanf("%d%d%d",&x,&r,&c);
    p=r*c;
    if(r>c)
        r=(r+c)-(c=r);
    if(x>=7)
      printf("Case #%d: RICHARD\n",q);
    else if(x==5 &&  (r==3 && c==5))
      printf("Case #%d: RICHARD\n",q);
    else if(x==6 && ((c==3) || (r==3)))
      printf("Case #%d: RICHARD\n",q);
    else if(x>p)
      printf("Case #%d: RICHARD\n",q);
    else if(p%x!=0)
      printf("Case #%d: RICHARD\n",q);
    else if(p%x==0 && r>=x-1 && c>=x-1)
      printf("Case #%d: GABRIEL\n",q);
    else if(p%x==0 && (r>=4 || c>=4) && x>=5)
      printf("Case #%d: GABRIEL\n",q);

    else if(p%x==0 && ((r==3 && c==10)) && x==5)
      printf("Case #%d: GABRIEL\n",q);
    else if(p%x==0 && ((r==3 && c==15)) && x==5)
      printf("Case #%d: GABRIEL\n",q);
    else if(p%x==0 && ((r==3 && c==20)) && x==5)
      printf("Case #%d: GABRIEL\n",q);

    else
      printf("Case #%d: RICHARD\n",q);
    q++;
    }
    return 0;
}
